Source/WebCore/layout/integration/inline/InlineIteratorTextBox.cpp

```diff
@@ -41,10 +41,29 @@ TextBoxIterator TextBox::nextTextBox() const
 
 LayoutRect TextBox::selectionRect(unsigned rangeStart, unsigned rangeEnd) const
 {
+    bool isCaretCase = rangeStart == rangeEnd;
+    rangeEnd = std::min(end(), rangeEnd);
+
     auto [clampedStart, clampedEnd] = selectableRange().clamp(rangeStart, rangeEnd);
 
-    if (clampedStart >= clampedEnd && !(rangeStart == rangeEnd && rangeStart >= start() && rangeStart <= end()))
-        return { };
+    if (clampedStart >= clampedEnd) {
+        if (!isCaretCase) {
```

Comment:
There is no obvious reason to use negated test here. Maybe just flip the branches around?

Reply:
Sure, fixed.